schema_salad/java_codegen.py

```diff
@@ -103,6 +103,54 @@ def _safe_makedirs(path):
         init="new AnyLoader()",
         loader_type="Loader<Object>",
     ),
+    "string": TypeDef(
```

Comment:
Great! Can you make some `_int_type_def` style intermediates like in the Python version to reduce the code duplication within `java_codegen.py`?